Tests/New-BBServerTag.Tests.ps1

```diff
@@ -0,0 +1,209 @@
+    }
+
+    return $conn
+}
+
+function New-TestProjectInfo
+{
+    $key = ([IO.Path]::GetRandomFileName()) -replace '[^A-Za-z0-9_]','_'
+    $key -replace '^\d+',''
+    'New-BBServerProject-New-Project-{0}' -f [IO.Path]::GetRandomFileName()
+}
+
+#setup
+Push-Location
+$conn = New-BBServerTestConnection
+$key,$repoName = New-TestProjectInfo
+$project = New-BBServerProject -Connection $conn -Key $key -Name $repoName -Description 'description'
+$repository = New-BBServerRepository -Connection $conn -ProjectKey $key -Name $repoName
+$cloneRepo = $repository.links.clone | Where-Object { $_.name -eq 'http' }  | Select-Object -ExpandProperty 'href'
+$global:commitNumber = 0
+
+#create netrc file to maintain credentials for commit and push
+$netrcFile = New-Item -Name '_netrc' -Force -Path $env:HOME -ItemType 'file' -Value @"
+machine $(([uri]$cloneRepo).Host)
+login $($conn.Credential.UserName)
+password $($conn.Credential.GetNetworkCredential().Password)
+"@
+
+<#
+    Don't forget that we've created a bunch of projects on BBServer that will likely need to be removed? Not sure about that 
+    though, as my local instance of BBserver probably doesn't have any effect on the outside world.
+
+    To access BBserver, get the login info from the _netrc file in the env:home directory. The browse URL is contained in the repo.links
+#>
+
+function GivenAValidCommit
+{
+    Set-Location $TestDrive.FullName
+    git init | Out-Null
+    #clone, commit and push a new file to the new repo
+    git clone $cloneRepo
+    git remote add origin $cloneRepo
+    git pull origin master | Out-Null
+    $newFile = New-Item -Name 'commitfile' -ItemType 'file' -Force -Value ('newFile {0}!!' -f $global:commitNumber) 
+    git add $newFile
+    git commit -m ('adding file, commit no {0} for project-key: {1}' -f $global:commitNumber++, $key) | Out-Null
+    
+    #get the HEAD commit hash
+    $commitHash = git rev-parse HEAD
+    
+    git push --set-upstream $cloneRepo master | Out-Null
+
+    return $commitHash
+
+}
+
+function WhenTaggingTheCommit
+{
+    param(
+        [String]
+        $CommitHash,
+
+        [String]
+        $TagName,
+
+        [String]
+        $Message
+    )
+
+    $optionalParams = @{}
+    if ($Message)
+    {
+        $optionalParams['Message'] = $Message
+    }
+    $global:Error.Clear()
+    try
+    {
+        New-BBServerTag -Connection $conn -ProjectKey $key -RepositoryKey $repoName -name $TagName -CommitID $commitHash @optionalParams
+    }
+    catch
+    {
+        return $false
+    }
+    return $true
+}
+
+function ThenTheCommitShouldBeTagged
+{
+    param(
+        $TagResult
+    )
+
+    it 'should not throw any errors' {
+        $Global:Error | Should BeNullOrEmpty
+    }
+    it 'should successfully tag the commit' {
+        $TagResult | Should be $true
+    }
+    <#
+    TODO: implement a way of checking to see that the remote repo was tagged with the appropriate tag.

```

Comment:
I think `git fetch --all` will bring down the tags so you can run `git tag` to view it. I think you could also run `git rev-parse TAG` to get the commit hash of the tag.